pkg/node/provisioners/utilwrappers/partition_operations.go

```diff
@@ -64,20 +66,28 @@ func (p *Partition) GetFullPath() string {
 // PartitionOperationsImpl is a base implementation for PartitionOperations interface
 type PartitionOperationsImpl struct {
 	ph.WrapPartition
-	log *logrus.Entry
+	log     *logrus.Entry
+	metrics metrics.Statistic
 }
 
 // NewPartitionOperationsImpl constructor for PartitionOperationsImpl and returns pointer on it
 func NewPartitionOperationsImpl(e command.CmdExecutor, log *logrus.Logger) *PartitionOperationsImpl {
+	var partMetrics = metrics.NewPartitionsMetrics()
+	if err := prometheus.Register(partMetrics.Collect()); err != nil {
+		log.WithField("component", "NewPartitionOperationsImpl").
+			Errorf("Failed to register metric: %v", err)
+	}
 	return &PartitionOperationsImpl{
 		WrapPartition: ph.NewWrapPartitionImpl(e, log),
 		log:           log.WithField("component", "PartitionOperations"),
+		metrics:       partMetrics,
 	}
 }
 
 // PreparePartition completely creates and prepares partition p on node
 // After that FS could be created on partition
 func (d *PartitionOperationsImpl) PreparePartition(p Partition) (*Partition, error) {
+	defer d.metrics.EvaluateDuration("PreparePartition", time.Now())
```

Comment:
My personal preference is to avoid passing `time.Now()` every time explicitly.
We can use something like [this](https://github.com/dell/csi-baremetal/blob/0a5b2d8911f196e562977e625cadcb08a6559d04/pkg/base/util/exec_time.go#L26
) to remove this need

Reply:
Done